### PR TITLE
update git installation on centos 7 to git 2.x

### PR DESCRIPTION
**Pull request by tomerd**
motivation: centos 7 ships with git 1.x which is too old for the toolchain usage

chages:
* use RHL software collections to install git 2.x
* add symlink since collections install in /opt (TBD better solution)

### CODE REVIEW

File: nightly-main/centos/7/Dockerfile [hunk 1]
```diff
@@ -20,6 +22,8 @@ RUN yum install shadow-utils.x86_64 -y \
   sqlite \
   zlib-devel
 
+RUN ln -s /opt/rh/rh-git227/root/bin/git /usr/bin/git
```
Comment (tomerd) [inline nightly-main/centos/7/Dockerfile:27]: I dont like this, and happy to change to a better way to achieve this. the other option of course is adding to scl-rh /opt location to the PATH but that also comes with its own set of issues

Comment (tomerd) [reply nightly-main/centos/7/Dockerfile:27]: @drexin @shahmishal wdyt?

**Comment by tomerd** [comment]
@shahmishal @drexin thoughts?

File: nightly-main/centos/7/Dockerfile (hunk 1 above)
Comment (shahmishal) [reply nightly-main/centos/7/Dockerfile:27]: Let's provide a comment explaining why we need git 2.x in the Dockerfile.

Comment (tomerd) [inline nightly-main/centos/7/Dockerfile:27]:
```suggestion
# centos 7 ships with git 1.x which is too old for the toolchain usage
RUN ln -s /opt/rh/rh-git227/root/bin/git /usr/bin/git
```

File: nightly-main/centos/7/Dockerfile [hunk 2]
```diff
@@ -2,10 +2,12 @@ FROM centos:7
 LABEL maintainer="Swift Infrastructure <swift-infrastructure@forums.swift.org>"
 LABEL description="Docker Container for the Swift programming language"
 
+RUN yum install -y centos-release-scl-rh
```
Comment (tomerd) [inline nightly-main/centos/7/Dockerfile:6]:
```suggestion
# centos 7 ships with git 1.x which is too old for the toolchain usage, using RH software collections to install git 2.x
RUN yum install -y centos-release-scl-rh
```

**Review by shahmishal — APPROVED** [review]

**Comment by stevapple** [comment]
It turns out that simply symlinking the executable won't work because `git` has runtime dependency on `httpd24`.

I managed to get it work by:
```dockerfile
RUN echo "source /opt/rh/rh-git227/enable" >> /etc/profile.d/sh.local
ENV ENV=/etc/profile.d/sh.local
```
though which the environment variables can be correctly set for both `bash` and `sh`. @tomerd @shahmishal

**Pull request merged**